pkg/didcomm/dispatcher/api.go

```diff
@@ -78,14 +80,63 @@ type DIDCommAction struct {
 	Callback Callback
 }
 
+// Result is used for notifying whether callback was finished
+// the result is successful when Err is nil
+type Result struct {
+	Err error
+}
+
 // DIDCommCallback message type to pass service callback in go channels.
 type DIDCommCallback struct {
 	// Set the value in case of any error while processing the DIDComm message event by the consumer.
-	Err error
+	res  chan Result
+	done bool
+}
+
+// NewDIDCommCallback creates DIDCommCallback message
+func NewDIDCommCallback() *DIDCommCallback {
+	return &DIDCommCallback{}
+}
+
+// WithResult creates result chan which provides the possibility to be notified when the callback is finished
+func (c *DIDCommCallback) WithResult() *DIDCommCallback {
+	if c.res != nil {
+		return c
+	}
+	c.res = make(chan Result)
```

Comment:
It seems we are forcing each Callback to have its own channel.

There is a trade-off here. (when channels are provided by the calling code, they can re-use them across many calls).

Reply:
I was thinking about reusing, but I did not like because when we receive an error we will never know for which callback it was.